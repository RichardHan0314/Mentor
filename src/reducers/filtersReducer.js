/* eslint-disable import/prefer-default-export */

import { SET_ALL_FILTERS, SET_FILTERS } from '../actions/filtersActions';

const initialState = {
    allFilters: [
        {id: 1, name: 'Action', status: false},
        {id: 2, name: 'Adventure', status: false},
        {id: 3, name: 'Animation', status: false },
        {id: 4, name: 'Drama', status: false},
        {id: 5, name: 'Romance', status: false},
        {id: 6, name: 'Family', status: false},
        {id: 7, name: 'Fantasy', status: false},
        {id: 8, name: 'Comedy', status: false},
    ],
    allFiltersChecked: true
};

export function filtersReducer(state = initialState, action) {
    switch (action.type) {
    case SET_ALL_FILTERS:
        return {...state, allFilters: action.payload.allFilters, allFiltersChecked: action.payload.allFiltersChecked };
    case SET_FILTERS:
        return {...state, allFilters: action.payload.allFilters, allFiltersChecked: action.payload.allFiltersChecked };
    default:
        return state;
    }
}