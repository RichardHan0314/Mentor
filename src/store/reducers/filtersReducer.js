/* eslint-disable import/prefer-default-export */
import { RESET_FILTERS, TOGGLE_FILTER } from '../actions/filtersActions';

const initialState = {
    filters: {
        action: false,
        adventure: false,
        animation: false,
        drama: false,
        romance: false,
        family: false,
        fantasy: false,
        comedy: false,
    }
};

export function filtersReducer(state = initialState, action) {
    switch (action.type) {
    case RESET_FILTERS:
        return {...state, filters: Object.keys(state.filters).reduce((acc, key) => ({ ...acc, [key]: false }), {}) };
    case TOGGLE_FILTER:
        return {
            ...state,
            filters: {
                ...state.filters,
                [action.payload.filter]: !state.filters[action.payload.filter],
            }
        };
    default:
        return state;
    }
}