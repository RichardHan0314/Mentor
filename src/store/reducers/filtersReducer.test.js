import { filtersReducer } from './filtersReducer';
import {
    RESET_FILTERS,
    TOGGLE_FILTER
} from '../actions/filtersActions';

describe('filters_reducer', () => {
    describe('INITIAL_STATE', () => {
        test('is correct', () => {
            const action = { type: 'dummy_action' };
            expect(filtersReducer(undefined, action)).toMatchSnapshot();
        });
    });
      
    describe('RESET_FILTERS', () => {
        test('returns the correct state', () => {
            const action = { type: RESET_FILTERS };

            expect(filtersReducer(undefined, action)).toMatchSnapshot();
        });
    });

    describe('TOGGLE_FILTER', () => {
        test('returns the correct state', () => {
            const filter = 'romance';
            const action = { type: TOGGLE_FILTER, payload: filter };

            expect(filtersReducer(undefined, action)).toMatchSnapshot();
        });
    });
});