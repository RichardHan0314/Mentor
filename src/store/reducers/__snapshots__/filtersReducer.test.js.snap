// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters_reducer INITIAL_STATE is correct 1`] = `
Object {
  "filters": Object {
    "action": false,
    "adventure": false,
    "animation": false,
    "comedy": false,
    "drama": false,
    "family": false,
    "fantasy": false,
    "romance": false,
  },
}
`;

exports[`filters_reducer RESET_FILTERS returns the correct state 1`] = `
Object {
  "filters": Object {
    "action": false,
    "adventure": false,
    "animation": false,
    "comedy": false,
    "drama": false,
    "family": false,
    "fantasy": false,
    "romance": false,
  },
}
`;

exports[`filters_reducer TOGGLE_FILTER returns the correct state 1`] = `
Object {
  "filters": Object {
    "action": false,
    "adventure": false,
    "animation": false,
    "comedy": false,
    "drama": false,
    "family": false,
    "fantasy": false,
    "romance": false,
    "undefined": true,
  },
}
`;
